### PR TITLE
Fix #4480 - show ACMG VUS on general report

### LINKED ISSUE #4480: Other commented variant missing from somatic general report?
User example is https://scout.scilifelab.se/cust087/F0042600-5/cancer/651bfc1bd6e834009ecfb701b7b5a1a2?cancer=yes. 
This one is edited and also looks ACMG-classified.

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

ACMG VUS have `acmg_classification == 0`. We are careful to use `"$exist"` when querying db, but this apparently slipped in checking on the variant - maybe in a refactor.. 

Screenshot 2024-03-07 at 08 47 53

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. ACMG classify a variant as "VUS"
2. check general report - it will be missing
3. optionally, also test e.g. commenting the same variant and find it still not showing
4. apply patch
5. find variant displaying on general case report

Screenshot 2024-03-07 at 08 59 07

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/cases/controllers.py [hunk 1]
```diff
@@ -594,7 +594,7 @@ def _get_decorated_var(var_obj: dict) -> dict:
         case_id=case_obj["_id"], institute_id=institute_obj["_id"]
     ):
         for eval_category, variant_key in CASE_REPORT_VARIANT_TYPES.items():
-            if var_obj.get(variant_key):
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/controllers.py:597]: That's because bool(0) is False?

Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:597]: Yes..

Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:597]: Ah, if I think about it, `None` values may be there as well. Hm, that should have been handled downstream, but maybe let me add an explicit None-check as well.

File: scout/server/blueprints/cases/controllers.py [hunk 2]
```diff
@@ -587,21 +573,48 @@ def _get_decorated_var(var_obj: dict) -> dict:
...
+        variant_obj=var_obj,
+        add_other=False,
+        get_overlapping=False,
+        variant_type=var_obj["category"],
+        institute_obj=institute_obj,
+        case_obj=case_obj,
+    )["variant"]
+
+
+def _partition_var_by_type(
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/controllers.py:604]:
```suggestion
def _set_evaluated_variants_by_type(
```

Feels more self-explanatory to me

File: scout/server/blueprints/cases/controllers.py [hunk 3]
```diff
@@ -587,21 +573,48 @@ def _get_decorated_var(var_obj: dict) -> dict:
...
-    # Collect all evaluated variants except causative, partial causative and suspected variants
     for var_obj in store.evaluated_variants(
         case_id=case_obj["_id"], institute_id=institute_obj["_id"]
     ):
-        for eval_category, variant_key in CASE_REPORT_VARIANT_TYPES.items():
-            if var_obj.get(variant_key):
-                evaluated_variants_by_type[eval_category].append(
-                    _get_decorated_var(var_obj=var_obj)
-                )
+        _partition_var_by_type(evaluated_variants_by_type, var_obj, institute_obj, case_obj)
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/controllers.py:583]:
```suggestion
        _set_evaluated_variants_by_type(evaluated_variants_by_type, var_obj, institute_obj, case_obj)
```

**Review by northwestwitch — APPROVED** [review]
Looks good! I have just a suggestion on the naming of a function

File: scout/server/blueprints/cases/controllers.py (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:604]: Hm, only one variant there so the suggested name would be a little misleading unless we wrap the whole loop in the function. Which might be a tad quicker, but these shouldn't be many. Set is generic, for good and bad, but point taken. I'll elaborate..

**Comment by dnil** [comment]
For reference, we've had this for about 5 months, from #4109. It's likely been a while for users to notice as mostly classified variants like that also go with a pin, which should have made them show.

**Comment by northwestwitch** [comment]
> For reference, we've had this for about 5 months

Yeah I guess VUS are not on top of the variant list they check, but good that now it is fixed! 🥇

**Pull request merged**